### DISCUSSION TITLE
Understanding errors from `into_pyobject()`

### DISCUSSION THREAD

**Discussion by CastilloDel** [PyO3/pyo3#4781] [Questions] [answered]
Hello! I was migrating some code to a newer version. In some place, I was using `.into_py(py)` to convert a Rust type (that uses the `[pyclass]` attribute) to a Python type. I saw this was deprecated and I should use `.into_pyobject(py)` instead. However, it was a surprise to me to find out that this returned a `Result` instead of just `Bound`. I checked https://pyo3.rs/v0.23.3/conversions/traits.html, but couldn't find any explanation about what could fail. 

TLDR: I would like to know if a struct that uses `[pyclass]` can really return an error when calling `into_pyobject()` and in which cases this could happen.

Thanks in advance!

**Comment by mejrs** [comment, marked as answer]
There are some cases where the conversion needs to import a type or call a python method, which can fail. Things like the date/time/hashmap etc related implementations are examples of conversions that can fail. On the other hand, if they cannot fail they use `Infallible` as the error type which means that on a new enough Rust you can write 
```rust
let Ok(x) = y.into_pyobject();
```

**Reply by CastilloDel** [reply]
Thanks for the answer!
